packages/core/src/agent/index.ts

```diff
@@ -1201,14 +1202,36 @@ export class Agent<
     threadId,
     memoryConfig,
     runId,
+    createThread,
   }: {
     saveQueueManager: SaveQueueManager;
     result: any;
     messageList: MessageList;
     threadId?: string;
     memoryConfig?: MemoryConfig;
     runId?: string;
+    createThread?: {
+      memory: MastraMemory;
+      resourceId: string;
+      threadId?: string;
+      title?: string;
+      metadata?: Record<string, unknown>;
+      memoryConfig?: MemoryConfig;
+      saveThread?: boolean;
+    };
   }) {
+    if (createThread) {
```

Comment:
I don't think it makes sense to create the thread inside this function, it's just a layer of indirection, it makes sense to just create where this function is called